### PR TITLE
Add heap dump files if the run failed

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Add heap dump files if the run failed

### Issues Resolved
https://github.com/opensearch-project/OpenSearch/issues/3973#issuecomment-1227886974

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Is this to help in debug java heap issues for gradle check?

**Comment by peterzhuamazon** [comment]
> Is this to help debug java heap issues for gradle check?

Yes:
https://github.com/opensearch-project/OpenSearch/issues/3973#issuecomment-1227886974

**Review by gaiksaya — APPROVED** [review]

File: jenkins/gradle/gradle-check.jenkinsfile [hunk 1]
```diff
@@ -129,6 +129,9 @@ pipeline {
                 }
             }
             post() {
+                failure {
+                    archiveArtifacts artifacts: 'search/java_pid*.hprof'
```
Comment (reta) [inline jenkins/gradle/gradle-check.jenkinsfile:133]: @peterzhuamazon could we change it please to  `archiveArtifacts artifacts: '**/build/heapdump/*.hprof', allowEmptyArchive: true`? thank you

Comment (peterzhuamazon) [reply jenkins/gradle/gradle-check.jenkinsfile]: You got it.

**Review by bbarani — APPROVED** [review]

**Review by reta — APPROVED** [review]

**Pull request merged**